tests/Browser/Pages/Backend/Categories/AdminListCategoriesTest.php

```diff
@@ -0,0 +1,170 @@
+        $categoryIds = DB::table('categories')->pluck('id')->toArray();
+
+        factory(Donator::class)->create();
+    
+        for ($i = 0; $i <= $rows; $i++)
+        {
+            factory(Book::class)->create([
+                'category_id' => $faker->randomElement($categoryIds),
+                'donator_id' => 1,
+                'name' => $faker->name,
+                'author' => $faker->name,
+            ]);
+        }
+    }
+
+    /**
+     * Check the list of categories without data
+     *
+     * @return void
+     */
+    public function testShowListCategoriesNoData()
+    {
+        $user = $this->userLogin();
+        $this->browse(function (Browser $browser) use ($user) {
+            $browser->loginAs($user)
+                    ->visit('/admin/categories/')
+                    ->resize(900, 1600)
+                    ->assertSee('List Categories');
+                    $elements = $browser->elements('#table-categories tbody tr');
+                    $this->assertCount(0, $elements);
+        });
+    }
+
+    /**
+     * Check list categories with showing only 10 rows
+     *
+     * @return void
+     */
+    public function testShowListCategoriesNoPagination()
+    {
+        $this->makeDataOfListCategories(8);
+        $user = $this->userLogin();
+        $this->browse(function (Browser $browser) use ($user) {
+            $browser->loginAs($user)
+                    ->visit('/admin/categories/')
+                    ->resize(900, 1600)
+                    ->assertSee('List Categories');
+                    $elements = $browser->elements('#table-categories tbody tr');
+                    $this->assertCount(8, $elements);
+        });
+    }
+
+    /**
+     * A Dusk test Pagination
+     *
+     * @return void
+     */
+    public function testPagination()
+    {
+        $this->makeDataOfListCategories(25);
+        $user = $this->userLogin();
+        $this->browse(function (Browser $browser) use ($user) {
+            $browser->loginAs($user)
+                    ->visit('/admin/categories/')
+                    ->resize(900, 1600)
+                    ->assertSee('List Categories');
+                    $elements = $browser->elements('.pagination li');
+                    $numberPage = count($elements) - 2;
+                    $this->assertTrue($numberPage == ceil(25 / (config('define.page_length'))));
+        });
+    }
+
+    /**
+     * Check list categories with showing than 10 rows
+     *
+     * @return void
+     */
+    public function testShowListCategoriesHavePagination()
+    {
+        $this->makeDataOfListCategories(15);
+        $user = $this->userLogin();
+        $this->browse(function (Browser $browser) use ($user) {
+            $browser->loginAs($user)
+                    ->visit('/admin/categories/')
+                    ->resize(900, 1600)
+                    ->assertSee('List Categories')
+                    ->click('.pagination li:nth-child(3) a');
```

Comment:
click song không biết đi đâu, em nên cho visit qua trang 2

Reply:
Các currents phía dưới current `click`:                  
                    ->click('.pagination li:nth-child(3) a');
                    $elements = $browser->elements('#table-categories tbody tr');
                    $this->assertCount(5, $elements);
                    $browser->assertQueryStringHas('page', 2);
                    $this->assertNotNull($browser->element('.pagination'));